app/src/main/java/net/squanchy/service/firebase/FirebaseDbService.kt

```diff
@@ -79,8 +80,12 @@ class FirebaseDbService(private val database: DatabaseReference) {
                     if (emitter.isDisposed) {
                         return
                     }
-                    val value = dataSnapshot.getValue(clazz)
-                    emitter.onNext(map(value))
+                    try {
+                        val value = dataSnapshot.getValue(clazz)
+                        emitter.onNext(map(value))
+                    } catch (e: Exception) {
+                        throw DatabaseException("Problem in DB at path $path, class $clazz", e)
```

Comment:
Does `emitter.onError` make sense here?

Reply:
Probably should yeah 👍

Reply:
Not 100% sure it will be equivalent since I was catching and wrapping a RuntimeException, but why not :man_shrugging: